.github/actions/pnpm/action.yml

```diff
@@ -3,12 +3,10 @@ description: Setup node and install dependencies using pnpm
 runs:
   using: "composite"
   steps:
-    - uses: pnpm/action-setup@v2.2.2
+    - uses: pnpm/action-setup@v4
+    - uses: actions/setup-node@v4
       with:
-        version: 7
```

Comment:
the updated action understands to look into `package.json#packageManager` to use the version defined there!